src/python/pants/backend/go/util_rules/cgo_test.py

```diff
@@ -169,3 +173,312 @@ def test_cgo_compile(rule_runner: RuleRunner) -> None:
         ],
     )
     assert result.stdout.decode() == "Hello World!\n"
+
+
+def _find_binary(binary_names: Iterable[str]) -> Path | None:
```

Comment:
Are these scanned out of the `PATH` in production usage? It might be good to add a TODO on a `Subsystem` or something pointing to https://github.com/pantsbuild/pants/issues/12203 to address the fact that we'll eventually want to reify locating those.

Reply:
They are scanned out of `--golang-cgo-tool-search-paths` in production. The intention in searching for them here is to be able to cleanly skip the test if the required binary is not available.